components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java

```diff
@@ -34,5 +36,46 @@
      * @throws NotificationTemplateManagerException Error while getting the notification template
      */
     NotificationTemplate getNotificationTemplate(String notificationChannel, String templateType, String locale,
-            String tenantDomain) throws NotificationTemplateManagerException;
+                                                 String tenantDomain) throws NotificationTemplateManagerException;
+
+    /**
+     * Add the notification template to the registry.
+     *
+     * @param notificationTemplate Notification template
+     *                             {@link org.wso2.carbon.identity.governance.model.NotificationTemplate}
+     * @param tenantDomain         Tenant domain
+     * @throws NotificationTemplateManagerException Error while adding the notification template.
+     */
+    void addNotificationTemplate(NotificationTemplate notificationTemplate, String tenantDomain)
+            throws NotificationTemplateManagerException;
+
+    /**
+     * Add a new notification template to the registry to the corresponding notification channel root directory.
+     *
+     * @param displayName         Notification template display name
+     * @param notificationChannel Notification channel
+     * @param tenantDomain        Tenant domain
+     * @throws NotificationTemplateManagerException Error adding the template to the registry.
+     */
+    void addNotificationTemplateType(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerException;
+
+    /**
+     * Add the default notification templates which matches the given notification channel to the respective tenants
+     * registry.
+     *
+     * @param notificationChannel Notification channel (Eg: SMS, EMAIL)
+     * @param tenantDomain        Tenant domain
+     * @throws NotificationTemplateManagerException Error adding the default notification templates
```

Comment:
Shall we change this doc comment to one like "Error while adding the template ..."

Reply:
Fixed: 948a27103b94bac935df64e1a165f77d7f4b5b91